packages/cactus-plugin-ledger-connector-corda/src/main/typescript/plugin-ledger-connector-corda.ts

```diff
@@ -243,4 +324,210 @@ export class PluginLedgerConnectorCorda
+          `Invalid holdingIDShortHash: ${req.holdingIDShortHash}`,
+        );
+      }
+      const getFlowUrl = urlcat(
+        this.baseUrl,
+        "flow/:holdingIDShortHash/:clientRequestId",
+        {
+          holdingIDShortHash: req.holdingIDShortHash,
+          clientRequestId: req.clientRequestId,
+        },
+      );
+      const response = await fetch(getFlowUrl, {
+        method: `GET`,
+        headers: headers,
+        agent,
+      });
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+      const responseData = await response.json();
+      return responseData;
+    } catch (error) {
+      console.error("Error fetching data:", error);
+    }
+  }
+
+  public async listFlows(req: GetFlowCidV1Request): Promise<any> {
+    try {
+      const { headers, agent } = await this.setupRequest(
+        req.username,
+        req.password,
+      );
+      if (typeof req.holdingIDShortHash === "undefined") {
+        throw new Error("holdingIDShortHash is undefined");
+      }
+      if (!this.validateHoldingIDShortHash(req.holdingIDShortHash)) {
+        throw new Error(
+          `Invalid holdingIDShortHash: ${req.holdingIDShortHash}`,
+        );
+      }
+      const listFlowsUrl = urlcat(this.baseUrl, "flow/:holdingIDShortHash", {
+        holdingIDShortHash: req.holdingIDShortHash,
+      });
+      const response = await fetch(listFlowsUrl, {
+        method: `GET`,
+        headers: headers,
+        agent,
+      });
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+      const responseData = await response.json();
+      return responseData;
+    } catch (error) {
+      console.error("Error fetching data:", error);
+    }
+  }
+
+  public async pollEndpointUntilCompleted(
+    shortHash: string,
+    clientRequestId: string,
+    interval = 5000,
+    maxAttempts = 10,
+  ): Promise<GetFlowCidV1Response> {
+    return new Promise<GetFlowCidV1Response>(async (resolve, reject) => {
+      let attempts = 0;
+
+      const poll = async () => {
+        attempts++;
+        const queryVar: GetFlowCidV1Request = {
+          username: "admin",
+          password: "admin",
+          holdingIDShortHash: shortHash,
+          clientRequestId,
+        };
+        try {
+          const response = await this.getFlow(queryVar);
+          if (response.flowStatus === "COMPLETED") {
+            resolve(response);
+          } else {
+            if (attempts < maxAttempts) {
+              setTimeout(poll, interval);
+            } else {
+              reject(
+                new Error(
+                  `Max attempts (${maxAttempts}) reached. Flow status not completed.`,
+                ),
+              );
+            }
+          }
+        } catch (error) {
+          if (attempts < maxAttempts) {
+            setTimeout(poll, interval);
+          } else {
+            reject(
+              new Error(
```

Comment:
@adrianbatuto Please throw `GatewayTimeoutError` here using the mentioned library in my other comment.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/504